### PR TITLE
(closes #1359) Fix pytest fixtures tear downs

### LINKED ISSUE #1359: Parallel pytest often fails with unititialised API constants
There are issues with uninitialised API constants when executing the tests in parallel, but serial unit test (like in the CI) or testing file by file do not produce any error.

I can replicate some error with:
`pytest -n 6 src/psyclone/tests/gocean1p0_test.py src/psyclone/tests/psyir/nodes/acc_directives_test.py`
so I assume the interference is between this 2 files.

### PR DESCRIPTION
**Pull request by sergisiso**

### CODE REVIEW

**Comment by sergisiso** [comment]
@rupertford @arporter @hiker This is ready to review. I recently had lots of spurious pytest failures when running the test suite in parallel, this PR should fix these problems.

**Comment by hiker** [comment]
I'll have a look at this tomorrow.

File: src/psyclone/tests/psyir/nodes/acc_directives_test.py [hunk 1]
```diff
@@ -49,11 +49,20 @@
 from psyclone.transformations import ACCLoopTrans, ACCEnterDataTrans, \
     ACCParallelTrans, ACCKernelsTrans
 from psyclone.tests.utilities import get_invoke
+from psyclone.configuration import Config
```
Comment (hiker) [inline src/psyclone/tests/psyir/nodes/acc_directives_test.py:52]: Could you sort the import alphabetically, please?

**Review by hiker — CHANGES_REQUESTED** [review]
I've verified that the bug existed before, and I could not reproduce a problem anymore (neither with the command specified #1359, nor by running all tests in parallel)

Just one minor change request, and you need to update the branch with latest master, then I am happy to merge.

File: src/psyclone/tests/psyir/nodes/acc_directives_test.py (hunk 1 above)
Comment (sergisiso) [reply src/psyclone/tests/psyir/nodes/acc_directives_test.py]: Done

**Comment by sergisiso** [comment]
@hiker Ready for another look

**Pull request merged**